platform/darwin/src/MGLTilePyramidOfflineRegion.mm

```diff
@@ -0,0 +1,66 @@
+#import "MGLTilePyramidOfflineRegion.h"
+
+#if !TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR
+    #import <Cocoa/Cocoa.h>
+#endif
+
+#import "MGLOfflineRegion_Private.h"
+#import "MGLGeometry_Private.h"
+#import "MGLStyle.h"
+
+@interface MGLTilePyramidOfflineRegion () <MGLOfflineRegion_Private>
+
+@end
+
+@implementation MGLTilePyramidOfflineRegion {
+    NSURL *_styleURL;
+}
+
+@synthesize styleURL = _styleURL;
+
+- (instancetype)init {
+    [NSException raise:@"Method unavailable"
+                format:
+     @"-[MGLTilePyramidOfflineRegion init] is unavailable. "
+     @"Use -initWithStyleURL:bounds:fromZoomLevel:toZoomLevel: instead."];
+    return nil;
+}
+
+- (instancetype)initWithStyleURL:(NSURL *)styleURL bounds:(MGLCoordinateBounds)bounds fromZoomLevel:(double)minimumZoomLevel toZoomLevel:(double)maximumZoomLevel {
+    if (self = [super init]) {
+        if (!styleURL) {
+            styleURL = [MGLStyle streetsStyleURL];
+        }
+        
+        if (!styleURL.scheme) {
+            // Assume a relative path into the application bundle.
+            styleURL = [NSURL URLWithString:[@"asset://" stringByAppendingString:styleURL.absoluteString]];
```

Comment:
Oh, that’s good to know. This line was copy-pasted from the MGLMapView implementation.

If a developer has a local style but wants to download online resources required by it, I suppose the workaround is to create a copy of the style in their Mapbox account and use it just for MGLTilePyramidOfflineRegion. I’ll document that workaround if you think it’s correct.